submodules/power_management_services_setup/variables.tf

```diff
@@ -16,7 +16,7 @@ variable "ssh_private_key" {
 
 variable "service_config" {
   description = "An object which contains configuration for NFS, NTP, DNS, Squid Services"
-  type        = map(any)
+  type        = any
```

Comment:
can you explain why this change was made?

Reply:
`variable "ntp_forwarder_config" {
  description = "Configuration for the NTP forwarder to an NTP service that is not reachable directly from PowerVS."
  type = object({
    ntp_enable        = bool
    server_host_or_ip = string
  })
  default = {
    "ntp_enable"        = "false"
    "server_host_or_ip" = ""
  }
}

variable "nfs_config" {
  description = "Configuration for the shared NFS file system (for example, for the installation media). Creates a filesystem of disk size specified, mounts and NFS exports it."
  type = object({
    nfs_enable        = bool
    server_host_or_ip = string
    nfs_file_system = list(object({
      name       = string
      mount_path = string
      size       = number
    }))
  })
  default = {
    "nfs_enable"        = "false"
    "server_host_or_ip" = ""
    "nfs_file_system"   = [{ name : "nfs", mount_path : "/nfs", size : 1000 }]
  }
}`

service_config: this variable takes in different config variables. Earlier nfs_filesystem was a string type. But now it is a list of object. if we use map(any) then all variables coming and getting assigned here must be of same type. Using  **any** there is no restriction. 
see here: 
https://github.com/terraform-ibm-modules/terraform-ibm-powervs-infrastructure/blob/d99494844d610813b01f64e1a670864a12a7ca6a/submodules/power_management_services_setup/main.tf#L99
https://github.com/terraform-ibm-modules/terraform-ibm-powervs-infrastructure/blob/d99494844d610813b01f64e1a670864a12a7ca6a/main.tf#L92